cdap-app-fabric/src/test/java/co/cask/cdap/scheduler/CoreSchedulerServiceTest.java

```diff
@@ -234,9 +268,14 @@ public void testRunScheduledJobs() throws Exception {
     int runs2 = getRuns(WORKFLOW_2);
     disableSchedule(AppWithFrequentScheduledWorkflows.DATASET_PARTITION_SCHEDULE_1);
     disableSchedule(AppWithFrequentScheduledWorkflows.DATASET_PARTITION_SCHEDULE_2);
+
     publishNotification(dataEventTopic, WORKFLOW_1, AppWithFrequentScheduledWorkflows.DATASET_NAME1);
+    MessageId messageId = waitUntilProcessed(dataEventTopic, null);
```

Comment:
It's possible that there were already some events processed before this notification is published? Then this`waitUntilProcessed` will immediately return. Maybe just call  `getAllJobs()` before publishing and record the number of jobs, and after publishing, call `getAllJobs()` again to wait for 3 more jobs?